internal/gatewayapi/validate.go

```diff
@@ -332,8 +333,12 @@ func (t *Translator) validateTLSConfiguration(listener *ListenerContext, resourc
 				)
 				break
 			}
+			var hostname string
+			if listener.Hostname != nil {
+				hostname = string(*listener.Hostname)
+			}
 
-			err := validateTLSSecretData(secret)
```

Comment:
thoughts on moving all the logic into the caller method so this API looks like `err := validateTLSSecretsData(secrets)` ?

Reply:
Right, this is a good idea to move them all inside tls.go.